src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.metrics;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+
+public class MetricsEmitter {
+    protected ScheduledFuture<?> future;
+    private final ScheduledExecutorService ses;
+    private final Object emitMetricsLock = new Object();
+    private static final int PERIODIC_AGGREGATE_INTERVAL = 3_600;
+    private final ClientDeviceAuthMetrics metrics;
+
+    /**
+     * Constructor for the MetricsEmitter.
+     *
+     * @param ses     {@link ScheduledExecutorService}
+     * @param metrics {@link ClientDeviceAuthMetrics}
+     */
+    @Inject
+    public MetricsEmitter(ScheduledExecutorService ses, ClientDeviceAuthMetrics metrics) {
+        this.ses = ses;
+        this.metrics = metrics;
+    }
+
+    /**
+     * Emit metrics once an hour.
+     */
+    public void emit() {
+        // Cancel previously running task
```

Comment:
nit: move this before the cancel call, otherwise it reads like this comment applies to the whole synchronized block